### ISSUE TITLE
SGD with decay

### ISSUE THREAD

**Issue by sumitpai** [Accenture/AmpliGraph#111]
**Background and Context**
SGD optimizer currently doesn't perform well on the tested datasets mainly due to the fixed learning rate. 
**Description**
Provide support to various tricks that can help boost the training performance (Eg: [SGD with warm restart](https://arxiv.org/abs/1608.03983) , decay at fixed intervals, etc)
